src/test/scala/io/citrine/lolo/bags/CorrelationStudy.scala

```diff
@@ -0,0 +1,440 @@
+package io.citrine.lolo.bags
+
+import breeze.stats.distributions.{RandBasis, ThreadLocalRandomGenerator}
+import io.citrine.lolo.{PredictionResult, TestUtils}
+import io.citrine.lolo.stats.StatsUtils.makeLinearCorrelatedData
+import io.citrine.lolo.stats.functions.Friedman
+import io.citrine.lolo.trees.multitask.MultiTaskTreeLearner
+import io.citrine.lolo.validation.{Merit, NegativeLogProbabilityDensity2d, StandardConfidence2d}
+import org.apache.commons.math3.random.MersenneTwister
+import org.knowm.xchart.{BitmapEncoder, XYChart}
+import org.knowm.xchart.BitmapEncoder.BitmapFormat
+
+import java.io.{BufferedWriter, File, FileWriter}
+import scala.util.Random
+
```

Comment:
I made heavy use of sealed traits (essentially enums) to codify the different options we have for running tests and to make it easy to add new options